### DISCUSSION TITLE
Is it possible to change the default page opened by `quarto preview`?

### DISCUSSION THREAD

**Discussion by nickvigilante** [quarto-dev/quarto-cli#9398] [Q&A] [answered]
### Description

We have a site with over 500 documents. When running `quarto preview` in a fresh environment with no cache, the site will usually render documents in a specific order, which I don't fully understand.

Anyway, we'd like to set the default page after `quarto preview` to always be a specific page in our repo, not just the last page Quarto renders. Is that possible? We're running version 1.4.553.

**Comment by mcanouil** [comment, marked as answer]
Run `quarto preview input.qmd` and possibly use `freeze`.

> not just the last page Quarto renders

I never seen this behaviour, Quarto either display the input from your `quarto preview` or when rendering/previewing a project, the "index" if there is one.
So, I'm not sure what you mean here as the "rendering order" is not what defines the page to be displayed in the preview.

At least in VSCode, when using the "Preview" button from the Quarto extension for VSCode, it does set the input which then open the preview to the matching page.
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/7539ea3a-fabd-436c-9dc4-425a94309de0">

**Reply by nickvigilante** [reply]
`quarto preview input.md` did the trick. It rendered the whole site from a clean cache and then opened the page. Thanks!

What I meant from my earlier comment was this: if I run just `quarto preview` against a project with a lot of files in it, there's a specific order that Quarto chooses to build each page.

For example, if I fork `quarto-web` locally and run `quarto preview`, it will compile things in `docs/projects` first, then `docs/blog`, then `docs/extensions`. I'm not sure how Quarto decided it would do it in that order:

```
[ 37/321] docs/projects/environment.qmd
[ 38/321] docs/projects/quarto-projects.qmd
[ 39/321] docs/projects/profiles.qmd
[ 40/321] docs/troubleshooting/index.qmd
[ 41/321] docs/books/book-basics.qmd
[ 42/321] docs/books/book-output.qmd
[ 43/321] docs/books/book-structure.qmd
[ 44/321] docs/books/book-crossrefs.qmd
[ 45/321] docs/blog/posts/2022-06-21-rstudio-conf-2022-quarto/index.qmd
[ 46/321] docs/blog/posts/2024-03-26-hugging-face/index.qmd
[ 47/321] docs/blog/posts/2023-04-26-1.3-release/index.qmd
[ 48/321] docs/blog/posts/2022-10-25-shinylive-extension/index.qmd
[ 49/321] docs/blog/posts/2022-07-25-feature-extensions/index.qmd
[ 50/321] docs/blog/posts/2024-01-24-1.4-release/index.qmd
[ 51/321] docs/blog/posts/2022-02-13-feature-callouts/index.qmd
...
[ 61/321] docs/blog/posts/2023-03-13-code-annotation/index.qmd
[ 62/321] docs/blog/posts/2023-12-05-asa-traveling-courses/index.qmd
[ 63/321] docs/blog/posts/2023-05-15-get-started/index.qmd
[ 64/321] docs/blog/index.qmd
[ 65/321] docs/extensions/listing-formats.qmd
[ 66/321] docs/extensions/filters.qmd
[ 67/321] docs/extensions/starter-templates.qmd
```

For some people in our org, the order in which the docs are generated is different.

**Reply by mcanouil** [reply]
I have to admit that I don't know how the order is defined but I am not sure it really matters.

**Reply by nickvigilante** [reply]
Yep, doesn't really matter, but it was confusing to our end users, so finding a way to make it operate consistently across all machines is what we were looking for. Thank you!
